### PR TITLE
added support for pex search

### PR DESCRIPTION
**Pull request by stepanbujnak**

### CODE REVIEW

File: pexae/client.py [hunk 1]
```diff
@@ -1,56 +1,53 @@
...
-                     ctypes.byref(c_status_code),
-                     c_status_message, c_status_message_size)
-        AEError.check(c_status_code.value, c_status_message.value.decode())
+    _lib.AE_Init(client_id.encode(), client_secret.encode(),
+                 ctypes.byref(c_status_code),
+                 c_status_message, c_status_message_size)
+    AEError.check(c_status_code.value, c_status_message.value.decode())
 
-        lock = _AE_Lock.new(_lib)
+    lock = _AE_Lock.new(_lib)
```
Comment (janderland) [inline pexae/client.py:29]: Where do you release this lock?

Comment (janderland) [reply pexae/client.py:29]: I'm seeing other locks below which don't seem to be released. When the lock goes out of scope does it get automatically released?

File: pexae/fingerprint.py [hunk 2]
```diff
@@ -28,52 +28,53 @@ def __init__(self, ft):
...
+
+        lk = _AE_Lock.new(_lib)
+
+        c_status = _AE_Status.new(_lib)
+        c_buf = _AE_Buffer.new(_lib)
+        c_ft = _AE_Buffer.new(_lib)
+
+        _lib.AE_Buffer_Set(c_buf.get(), buf, len(buf))
+
+        # TODO
```
Comment (janderland) [inline pexae/fingerprint.py:72]: This `TODO` can be removed.

**Review by janderland — APPROVED** [review]

File: pexae/client.py (hunk 1 above)
Comment (stepanbujnak) [reply pexae/client.py:29]: Yes, that's how it works: https://github.com/Pexeso/ae-sdk-py/blob/main/pexae/lib.py#L36 the unlock is called in the destructor.

**Pull request merged**